hail/python/hailtop/aiocloud/aiogoogle/client/storage_client.py

```diff
@@ -333,6 +333,11 @@ async def bucket_info(self, bucket: str) -> Dict[str, Any]:
         self._update_params_with_user_project(kwargs, bucket)
         return await self.get(f'/b/{bucket}', **kwargs)
 
+    async def object_info(self, bucket: str, name: str) -> Dict[str, Any]:
+        kwargs: Dict[str, Any] = {}
+        self._update_params_with_user_project(kwargs, bucket)
+        return await self.get(f'/b/{bucket}/o/{urllib.parse.quote(name, safe="")}', **kwargs)
```

Comment:
I think you can use `statfile` and add a `storage_class` field on `GetObjectFileStatus`